lib/validate_doc_update.js

```diff
@@ -72,25 +72,6 @@ var checkAuthority = function(newDoc, oldDoc, userCtx, secObj) {
   if (userCtx.facility_id === newDoc._id) {
     _err('You are not authorized to edit your own place');
   }
-
-  // district admins can do almost anything
-  if (hasRole(userCtx, 'district_admin')) {
-    return;
-  }
-
-  // only admins can delete
-  if (oldDoc && newDoc._deleted) {
-      _err('You must be an admin to delete docs');
-  }
-
-  // gateway and data_entry users can update
-  if (hasRole(userCtx, 'kujua_gateway') ||
-      hasRole(userCtx, 'data_entry')) {
-    return;
-  }
-
-  // none of the above
-  _err('You must be an admin, gateway, or data entry user to edit documents');
```

Comment:
It leaves a hole for a role like "analytics" which could now edit docs. In 3.0.0 we're looking at doing doc-level permissions which will close this hole again. I'm comfortable with this because there aren't many accounts that use a read only role.

This function is incredibly limited so we can't do things like access permissions or other configuration so in future this sort of functionality will be implemented in API. We could look at blocking writing to docs for read only users at the API level for 2.16.0 but it's not easy to find all the CouchDB APIs that end up modifying docs.

What do you think?